internal_utils.go

```diff
@@ -101,11 +102,23 @@ func newChannelContext(ctx context.Context, options ...func(builder *contextBuil
 
 // GetWorkerIdentity gets a default identity for the worker.
 func getWorkerIdentity(tasklistName string) string {
+	return fmt.Sprintf("%d@%s@%s", os.Getpid(), getHostName(), tasklistName)
+}
+
+func getHostName() string {
 	hostName, err := os.Hostname()
 	if err != nil {
 		hostName = "UnKnown"
 	}
-	return fmt.Sprintf("%d@%s@%s", os.Getpid(), hostName, tasklistName)
+	return hostName
+}
+
+// worker uuid per process
+var workerUUID string = uuid.New()
+
+func getWorkerTaskList(tasklist string) string {
+	// includes original tasklist and hostname for debuggability, workerUUID guarantees the uniqueness
```

Comment:
Why is it task list specific? We can have more than one global task list polled by a process. I don't think it means that we need to have multiple worker specific task lists. Having that number of workers can be potentially large doubling or even multiplying number of worker specific task lists could be not a good idea.

Reply:
Are you concerning that we have too many task list? I think the task list is super cheap on server. Our currently use case are all having one task list per worker host. But I agree that we could have one worker specific task list for multiple task lists. I can remove the task list from here if that is your concern.